### ISSUE TITLE
Doubt regarding validation and test set size

### ISSUE THREAD

**Issue by vandana-rajan** [Sunner4nwpu/TEMMA#1]
Hi,

Thanks for the code.

In your paper, you mentioned that the training data is converted into sequence format with sequences of 30s long and 10s overlap.
So, the training data input would be of size (batch_size, seq_length, feature_dimension), where seq_length=750 (i.e, 30/0.04). Now, in your test set, are you doing the same or is the test set size equal to (batch_size, feature_dimension)? 

Thanks,
VR

**Comment by Sunner4nwpu** [comment]
Thanks, this data augment is only implemented for the training set, as to the development set and test set we use the original length of each video, i.e, if a video has 1min long, the data input would be of size (batch_size=1, seq_length=1500 (60/0.04), feature_dimension)

**Comment by vandana-rajan** [comment]
Thanks for the quick reply.

**Comment by vandana-rajan** [comment]
Hi,

Sorry. I have one more doubt.

For augmentation, how did you select the sequence length of 30 seconds and overlap of 10 seconds? Was it random or is there any explanation for the numbers 30 and 10? Also, for selecting the validation set, you mentioned that out of 9 speakers in training, you took 1 out and repeated it 3 times. How did you select the validation set speakers? Was that random?

**Comment by Sunner4nwpu** [comment]
Thanks for your good question.
(1) Ref.[67] in the paper gives some experiments for different data augmentation, actually I have tried different parameters for data augmentation, I upload a demo dataset.py which may help you with easy setting different parameters for data augmentation. I select the validation set speacker by random, I suggest you'd better not to split the training set if you could access the test set.
(2) The post-processing operation provided by AVEC2016 baseline system is used, and indeed most papers that use RECOLA including the results for comparison in the paper employ this post-processing operation, but this is only done for RECOLA  but not for SEWA (AVEC2019)

**Comment by Sunner4nwpu** [comment]
Thanks for your good question.
(1) Ref.[67] in the paper gives some experiments for different data augmentation, actually I have tried different parameters for data augmentation, I upload a demo dataset.py which may help you with easy setting different parameters for data augmentation. I select the validation set speacker by random, I suggest you'd better not to split the training set if you could access the test set.
(2) The post-processing operation provided by AVEC2016 baseline system is used, and indeed most papers that use RECOLA including the results for comparison in the paper employ this post-processing operation, but this is only done for RECOLA  but not for SEWA (AVEC2019)

**Comment by vandana-rajan** [comment]
Okay. Understood. Thanks a lot for replying.
